disco/models/upgrade_cost_analysis_generic_model.py

```diff
@@ -0,0 +1,360 @@
+import logging
+from pathlib import Path
+from typing import List, Optional, Set
+
+from pydantic import BaseModel, Field, root_validator
+
+from jade.utils.utils import load_data
+from PyDSS.controllers import PvControllerModel
+from disco.models.base import BaseAnalysisModel
+
+
+logger = logging.getLogger(__name__)
+
+
+class UpgradeParamsBaseModel(BaseModel):
+    """Base model for all upgrade cost analysis parameters"""
+
+    class Config:
+        title = "UpgradeParamsBaseModel"
+        anystr_strip_whitespace = True
+        validate_assignment = True
+        validate_all = True
+        extra = "forbid"
+        use_enum_values = False
+
+
+class ThermalUpgradeParamsModel(UpgradeParamsBaseModel):
+    """Thermal Upgrade Parameters for all jobs in a simulation"""
+
+    xfmr_upper_limit: float = Field(
+        title="xfmr_upper_limit",
+        description="Transformer upper limit",
+        default=1.25,
+    )
+    line_upper_limit: float = Field(
+        title="line_upper_limit",
+        description="Line upper limit",
+        default=1.25,
+    )
+    line_design_pu: float = Field(
+        title="line_design_pu",
+        description="Line design P.U.",
+        default=0.75,
+    )
+    xfmr_design_pu: float = Field(
+        title="xfmr_design_pu",
+        description="Transformer design P.U.",
+        default=0.75,
+    )
+    voltage_upper_limit: float = Field(
+        title="voltage_upper_limit",
+        description="Voltage upper limit",
+        default=1.05,
+    )
+    voltage_lower_limit: float = Field(
+        title="voltage_lower_limit",
+        description="Voltage lower limit",
+        default=0.95,
+    )
+
+# TODO DT: document error codes
+# document opendss errors specifically
+
+class VoltageUpgradeParamsModel(UpgradeParamsBaseModel):
```

Comment:
FYI. Sherin told the upgrade params (thermal/voltage) may change, if happened, we'll update the model as well.